pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -1404,6 +1405,8 @@ public void uploadToDeepStoreIfMissing(TableConfig tableConfig, List<SegmentZKMe
           retentionMs - MIN_TIME_BEFORE_SEGMENT_EXPIRATION_FOR_FIXING_DEEP_STORE_COPY_MILLIS);
     }
 
+    PinotFS pinotFS = createPinotFS(rawTableName);
```

Comment:
PinotFS is not per table but per Pinot cluster. It is a bit strange to create the PinotFS from a table name. After looking through the two private methods added, it is much more clearer if we can get the PinotFS based on the scheme part of  _controllerConf.getDataDir().